xchange-stream-ftx/src/main/java/info/bitrich/xchangestream/ftx/FtxStreamingTradeService.java

```diff
@@ -0,0 +1,30 @@
+package info.bitrich.xchangestream.ftx;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import info.bitrich.xchangestream.core.StreamingTradeService;
+import io.reactivex.Observable;
+import org.knowm.xchange.currency.CurrencyPair;
+import org.knowm.xchange.dto.trade.UserTrade;
+
+public class FtxStreamingTradeService implements StreamingTradeService {
+
+  private final FtxStreamingService service;
+  private final Observable<JsonNode> fills;
+
+  public FtxStreamingTradeService(FtxStreamingService service) {
+    this.service = service;
+    this.fills = service.subscribeChannel("fills");
+  }
+
+  @Override
+  public Observable<UserTrade> getUserTrades(CurrencyPair currencyPair, Object... args) {
+
+    return fills
+        .filter(
```

Comment:
```
   fills.filter(jsonNode -> jsonNode.hasNonNull("data"))
         .map(jsonNode -> new CurrencyPair(jsonNode.get("data").get("market").asText()))
         .filter(cp -> cp.equals(currencyPair));
```

rewrite to clean up lambdas a bit

Reply:
I don't get you in this one, what do you want to do? If i copy paste your code it doesn't work.

Reply:
instead of doing so much in a single filter, create one filter and one map that breaks each piece of logic in the lambda into single lines, it should be easier to read